cocos2d/CCLabelTTF.m

```diff
@@ -645,289 +644,252 @@ - (CCTexture*) createTextureWithAttributedString:(NSAttributedString*)attributed
-    {
-        GLKVector4 outlineColor = _outlineColor.glkVector4;
-        
-        CGContextSetTextDrawingMode(context, kCGTextFillStroke);
-        CGContextSetRGBStrokeColor(context, outlineColor.r, outlineColor.g, outlineColor.b, outlineColor.a);
-        CGContextSetRGBFillColor(context, outlineColor.r, outlineColor.g, outlineColor.b, outlineColor.a);
-        CGContextSetLineWidth(context, outlineWidth * 2);
-        CGContextSetLineJoin(context, kCGLineJoinRound);
-        
-        [self drawString:string withFont:font inContext:context inRect:drawArea];
-        
-        // Don't draw shadow for main font
-        CGContextSetShadowWithColor(context, CGSizeZero, 0, NULL);
-        
-        if (hasShadow)
-        {
-            // Draw again, because shadows overlap
-            [self drawString:string withFont:font inContext:context inRect:drawArea];
-        }
-        
-        CGContextSetTextDrawingMode(context, kCGTextFill);
-    }
-    
-    // Handle font color
-    
-    CGColorSpaceRef colorspace = CGColorSpaceCreateDeviceRGB();
-    const CGFloat components[] = {_fontColor.red, _fontColor.green, _fontColor.blue, _fontColor.alpha};
-    CGColorRef color = CGColorCreate(colorspace, components);
-    CGColorSpaceRelease(colorspace);
-
-    CGContextSetFillColorWithColor(context, color);
-    CGContextSetStrokeColorWithColor(context, color);
-    CGColorRelease(color);
-
-    [self drawString:string withFont:font inContext:context inRect:drawArea];
-
-#if __CC_PLATFORM_IOS
-    UIGraphicsPopContext();
-#endif
-    CGContextRelease(context);
-
-    CCTexture* texture = NULL;
-
-    // Initialize the texture
-    if (useFullColor)
-    {
-        // RGBA8888 format
-        texture = [[CCTexture alloc] initWithData:data pixelFormat:CCTexturePixelFormat_RGBA8888 pixelsWide:POTSize.width pixelsHigh:POTSize.height contentSizeInPixels:dimensions contentScale:[CCDirector sharedDirector].contentScaleFactor];
-        [texture setPremultipliedAlpha:YES];
-    }
-    else
-    {
-        NSUInteger textureSize = POTSize.width * POTSize.height;
-        
-        // A8 format (alpha channel only)
-        unsigned char* dst = data;
-        for(int i = 0; i<textureSize; i++)
-            dst[i] = data[i*4+3];
-        
-        texture = [[CCTexture alloc] initWithData:data pixelFormat:CCTexturePixelFormat_A8 pixelsWide:POTSize.width pixelsHigh:POTSize.height contentSizeInPixels:dimensions contentScale:[CCDirector sharedDirector].contentScaleFactor];
-        self.shader = [CCShader positionTextureA8ColorShader];
-    }
-
-    free(data);
-    CFRelease(font);
-
-    return texture;
-}
-
-#endif
+    CCImage *image = [[CCImage alloc] initWithPixelSize:POTSize contentScale:[CCDirector sharedDirector].contentScaleFactor pixelData:pixelData];
+    image.contentSize = CC_SIZE_SCALE(dimensions, 1.0/image.contentScale);
+	return [[CCTexture alloc] initWithImage:image options:nil];
+}
+
+
+//#pragma mark -
+//#pragma mark Render Font iOS 5
```

Comment:
That was meant to be removed actually. We stopped officially supporting iOS 5 sometime last summer?